odoo/tools/translate.py

```diff
@@ -1316,22 +1355,26 @@ def __init__(self, cr, modules=None, lang=None):
 
     def _export_translatable_records(self):
         """ Export translations of all translated records having an external id """
+        modules = self._installed_modules if 'all' in self._modules else list(self._modules)
+        xml_defined = set()
+        for module in modules:
+            for path in get_xml_paths(module, self.env):
+                with file_open(path, mode='rb') as source:
+                    for entry in XMLFileReader(source, module=module):
+                        xml_defined.add((entry['imd_model'], entry['imd_name']))
```

Comment:
I am ok since it works. But I prefer not making the definition of exportable translation tricky.

In this PR: the translation of a field can be exported when its record has external id and there is no translation for any field of the record defined in the xml.

I propose: using an extra column in `ir_model_data.export_translation` to explicitly define if the external id should be used to export translation. And you can use `<record id="tax_report" model="account.report" export_translation="False">` to explicity avoid translations being exported.
Then the definition becomes: the translation of a field can be exported when its record have an external id with an export_translation=True.
As a result, we don't have to read xml files here when export.